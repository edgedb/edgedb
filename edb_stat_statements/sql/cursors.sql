--
-- Cursors
--

-- These tests require track_utility to be enabled.
SET edb_stat_statements.track_utility = TRUE;
SELECT edb_stat_statements_reset() IS NOT NULL AS t;

-- DECLARE
-- SELECT is normalized.
DECLARE cursor_stats_1 CURSOR WITH HOLD FOR SELECT 1;
CLOSE cursor_stats_1;
DECLARE cursor_stats_1 CURSOR WITH HOLD FOR SELECT 2;
CLOSE cursor_stats_1;

SELECT calls, rows, query FROM edb_stat_statements ORDER BY query COLLATE "C";
SELECT edb_stat_statements_reset() IS NOT NULL AS t;

-- FETCH
BEGIN;
DECLARE cursor_stats_1 CURSOR WITH HOLD FOR SELECT 2;
DECLARE cursor_stats_2 CURSOR WITH HOLD FOR SELECT 3;
FETCH 1 IN cursor_stats_1;
FETCH 1 IN cursor_stats_2;
CLOSE cursor_stats_1;
CLOSE cursor_stats_2;
COMMIT;

SELECT calls, rows, query FROM edb_stat_statements ORDER BY query COLLATE "C";
SELECT edb_stat_statements_reset() IS NOT NULL AS t;
