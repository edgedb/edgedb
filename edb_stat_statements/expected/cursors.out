--
-- Cursors
--
-- These tests require track_utility to be enabled.
SET edb_stat_statements.track_utility = TRUE;
SELECT edb_stat_statements_reset() IS NOT NULL AS t;
 t 
---
 t
(1 row)

-- DECLARE
-- SELECT is normalized.
DECLARE cursor_stats_1 CURSOR WITH HOLD FOR SELECT 1;
CLOSE cursor_stats_1;
DECLARE cursor_stats_1 CURSOR WITH HOLD FOR SELECT 2;
CLOSE cursor_stats_1;
SELECT calls, rows, query FROM edb_stat_statements ORDER BY query COLLATE "C";
 calls | rows |                         query                         
-------+------+-------------------------------------------------------
     2 |    0 | CLOSE cursor_stats_1
     2 |    0 | DECLARE cursor_stats_1 CURSOR WITH HOLD FOR SELECT $1
     1 |    1 | SELECT edb_stat_statements_reset() IS NOT NULL AS t
(3 rows)

SELECT edb_stat_statements_reset() IS NOT NULL AS t;
 t 
---
 t
(1 row)

-- FETCH
BEGIN;
DECLARE cursor_stats_1 CURSOR WITH HOLD FOR SELECT 2;
DECLARE cursor_stats_2 CURSOR WITH HOLD FOR SELECT 3;
FETCH 1 IN cursor_stats_1;
 ?column? 
----------
        2
(1 row)

FETCH 1 IN cursor_stats_2;
 ?column? 
----------
        3
(1 row)

CLOSE cursor_stats_1;
CLOSE cursor_stats_2;
COMMIT;
SELECT calls, rows, query FROM edb_stat_statements ORDER BY query COLLATE "C";
 calls | rows |                         query                         
-------+------+-------------------------------------------------------
     1 |    0 | BEGIN
     1 |    0 | CLOSE cursor_stats_1
     1 |    0 | CLOSE cursor_stats_2
     1 |    0 | COMMIT
     1 |    0 | DECLARE cursor_stats_1 CURSOR WITH HOLD FOR SELECT $1
     1 |    0 | DECLARE cursor_stats_2 CURSOR WITH HOLD FOR SELECT $1
     1 |    1 | FETCH 1 IN cursor_stats_1
     1 |    1 | FETCH 1 IN cursor_stats_2
     1 |    1 | SELECT edb_stat_statements_reset() IS NOT NULL AS t
(9 rows)

SELECT edb_stat_statements_reset() IS NOT NULL AS t;
 t 
---
 t
(1 row)

